src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,46 @@
 package core.basesyntax;
 
-class Pair {
+import java.util.Objects;
 
+class Pair<T1, T2> {
+    private T1 first;
+    private T2 second;
+
+    private Pair(T1 first, T2 second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    static <T1, T2> Pair of(T1 first, T2 second) {
+
```

Comment:
```suggestion
```

Reply:
I've made constructor private and fabric public static, are those correct?